src/main/java/duke/storage/Storage.java

```diff
@@ -0,0 +1,128 @@
+package duke.storage;
+
+import duke.tasklist.TaskList;
+import duke.exceptions.DukeException;
+import duke.tasks.Deadline;
+import duke.tasks.Event;
+import duke.tasks.Task;
+import duke.tasks.ToDo;
+
+import java.io.File;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.ArrayList;
+import java.util.Scanner;
+
+public class Storage {
+
+    private final String filePath;
+
+    public Storage(String input) {
+        this.filePath = input;
+    }
+
+    /**
+     * Reads existing task text file from local folder
+     * If file does not exist, create new text file
+     */
+    public void readFile() {
+        try {
+            Path dataFilePath = Paths.get("data/");
+            Files.createDirectories(dataFilePath);
+            File file = new File(this.filePath);
+
+            if (file.createNewFile()) {
+                System.out.println("New file created at " + filePath);
+            }
+        } catch (IOException e) {
+            System.out.println("Unable to create file " + filePath);
+        }
+    }
+
+    /**
+     * Save current task list into a text file
+     *
+     * @param tasks the list of tasks
+     * @throws DukeException if TaskList instance from user is invalid
+     */
+    public void saveData(TaskList tasks) {
+        try {
+            FileWriter fw = new FileWriter(filePath);
+
+            for (Task currTask : tasks.getTasks()) {
+                String description = currTask.getDescription();
+                int isComplete = currTask.isComplete() ? 1 : 0;
+                String saveText;
+
+                if (currTask instanceof ToDo) {
+                    saveText = "T | " + isComplete + " | " + description;
+                } else if (currTask instanceof Deadline) {
+                    saveText = "D | " + isComplete + " | " + description + " | " + ((Deadline) currTask).getDeadlineBy();
+                } else if (currTask instanceof Event) {
+                    saveText = "E | " + isComplete + " | " + description + " | " + ((Event) currTask).getEventBy();
+                } else {
+                    fw.close();
```

Comment:
Very neat on this part!
Similar to before, could this be transferred to the ui.java class?

Still, good work!